Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/components/batteryInput/index.tsx

```diff
@@ -1,4 +1,23 @@
-const BatteryInput = () => {
+interface BatteryInputProps {
+  setInput: React.Dispatch<
+    React.SetStateAction<{
+      batteryInput: number;
+      speedInput: number;
+      weatherInput: number;
+    }>
+  >;
+  setShow: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export function BatteryInput({ setInput, setShow }: BatteryInputProps) {
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
in every handleChange function, you do setShow(false);
this is a lot of duplicate code, can you use useReducer to see if you can simplify the code even more?